pkg/gstreamer/bin.go

```diff
@@ -381,6 +385,38 @@ func (b *Bin) sendEOS() {
 	}
 }
 
+func (b *Bin) AddOnEOSReceived(f func()) error {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	if len(b.elements) == 0 {
+		return nil
+	}
+
+	sink := b.elements[len(b.elements)-1]
```

Comment:
Can we add a comment about why the sink is guaranteed to be the last element in the array?

Reply:
It isn't, but it doesn't matter - what this is actually doing is checking that the EOS reached the last element